schema/table_test.go

```diff
@@ -56,17 +214,32 @@ func TestTableExecution(t *testing.T) {
 		t.Run(tc.Table.Name, func(t *testing.T) {
 			m := testClient{}
 			resources := make(chan *Resource)
+			summary := SyncSummary{}
 			go func() {
 				defer close(resources)
-				tc.Table.Resolve(ctx, m, nil, resources)
+				summary = tc.Table.Resolve(ctx, m, nil, resources)
 			}()
-			var i = 0
+			var i = uint64(0)
 			for resource := range resources {
 				if reflect.DeepEqual(resource.Data, tc.Resources[i].Data) {
 					t.Errorf("expected %v, got %v", tc.Resources[i].Data, resource)
 				}
 				i++
 			}
+			if tc.Summary.Resources != i {
+				t.Errorf("expected %d resources, got %d", tc.Summary.Resources, i)
+			}
+			if tc.Summary.Resources != summary.Resources {
+				t.Errorf("expected %d summary resources, got %d", tc.Summary.Resources, summary.Resources)
+			}
```

Comment:
nit: Looks like the second comparison here can never fail except if either: the case above it already failed, or the test is set up incorrectly. I'd probably remove the `!= i` check

Reply:
it can (It did and this is why I've added it :) ) because the first check checks how many resources were returned on the channel and the second one just checks the statistics struct

Reply:
Oh I see now, sorry yeah. It's because it's written in `want != got` instead of `got != want` order that I misread it. My nit would then be to change the ordering :)

Reply:
I think it is usually `expected != got` but also sometimes it's hard to keep the order consistent because for example you can't do `1 != some_var` so you still have constraints.

Reply:
In Go it's usually `got != expected`, so that it's consistent when you have a constant `got != 1` as well. https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures